### ISSUE TITLE
onSelectNotification runs whenever it enters the widget

### ISSUE THREAD

**Issue by LuisCampos97** [MaikuB/flutter_local_notifications#985]
Hi!
I have this code:
```dart
@override
  void initState() {
    super.initState();

    flutterLocalNotificationsPlugin = FlutterLocalNotificationsPlugin();
    final android = AndroidInitializationSettings('@mipmap/icon_fact');
    final iOS = IOSInitializationSettings();
    final initSettings = InitializationSettings(android: android, iOS: iOS);

    flutterLocalNotificationsPlugin.initialize(initSettings, onSelectNotification: _onSelectNotification);
  }

  Future<void> _onSelectNotification(String json) async {
    print(json);
    final obj = jsonDecode(json);

    if (obj['isSuccess']) {
      OpenFile.open(obj['filePath']);
    } else {
      showDialog(
          context: context,
          builder: (_) => AlertDialog(
                title: Text('Error'),
                content: Text('${obj['error']}'),
              ));
    }
  }
```

The first time I enter the widget it works well, when I click on the notification it triggers the _onSelectNotification function and does that OpenFile.open () that opens a file. 
But when I leave the widget and enter again, it always enters the _onSelectNotification function without me clicking on any notification. I just wanted it to trigger the _onSelectNotification function when I clicked on a notification.

Can you help me please?
Thank you,
Luís

**Comment by MaikuB** [comment]
This is to do with part I've documented [here](https://github.com/MaikuB/flutter_local_notifications/tree/master/flutter_local_notifications#initialisation) in making sure the plugin is only initialised once and doing it multiple times (even if it's a new instance) can cause the issue you're seeing.

Given that there's the ability to get the notification that has launched the app, there is some overlap here so if can help out with submitting a PR to address this behaviour then I'd be happy to review this

**Comment by AdelKanso** [comment]
@MaikuB I only initialize locale notification once, and it still runs when entering the page

**Comment by MaikuB** [comment]
@AdelKanso I'm referring to initialising once in the entire lifetime of the application even if the instance of the plugin that was created is scoped to that page. If you can back and forth to the same page where `initialize()` gets called then this is still the same as what I'm referring to. This has changed in the 4.0 prerelease versions so I'd suggest trying them out as it would be good to get feedback on if that works for others

**Comment by AdelKanso** [comment]
@MaikuB  Thank you it works for me now.

**Comment by MaikuB** [comment]
@AdelKanso do you mean you use the 4.0 prerelease or you made changes in your code?

**Comment by AdelKanso** [comment]
I made changes to my code, I've initialize locale notification in initState
of Home page, i was initializing it in didChangedependencies in Home page,
and it was invoked multiple times, wich made the error

**Comment by Sandeep19031** [comment]
> @AdelKanso I'm referring to initialising once in the entire lifetime of the application even if the instance of the plugin that was created is scoped to that page. If you can back and forth to the same page where `initialize()` gets called then this is still the same as what I'm referring to. This has changed in the 4.0 prerelease versions so I'd suggest trying them out as it would be good to get feedback on if that works for others

yes This is doing right. flutter_local_notifications: ^4.0.0-dev.2

**Comment by MaikuB** [comment]
Thanks @Sandeep19031. I've published 4.0.0 as a stable release now
